src/components/Toaster/Toaster.tsx

```diff
@@ -0,0 +1,111 @@
+    actions?: ActionProps[];
+    /** Has a close button, default: true */
+    isClosable?: boolean;
+    /** A label for the close button that will be used by screenreaders */
+    closeButtonLabel?: string;
+}
+
+const { block, elem } = bem('Toast', styles);
+
+const ContextIcon = ({ context }: { context: Context }) => {
+    switch (context) {
+        case 'info':
+            return <InfoIcon data-testid="info-icon" />;
+        case 'success':
+            return <SuccessIcon data-testid="success-icon" />;
+        case 'cautious':
+            return <CautiousIcon data-testid="cautious-icon" />;
+        case 'critical':
+            return <CriticalIcon data-testid="critical-icon" />;
+        default:
+            return assertNever(context, 'Toaster.tsx');
+    }
+};
+
+export const toast = ({
+    title,
+    description,
+    context = 'info',
+    actions,
+    isClosable = true,
+    closeButtonLabel,
+}: ToastProps) =>
+    sonnerToast.custom(
+        (t) => (
+            <div {...block({ context })}>
+                <div {...elem('contextIcon')}>
+                    <ContextIcon context={context} />
+                </div>
+                <div {...elem('content')}>
+                    <div {...elem('message')}>
+                        <h3 className="OneUI-heading-3">{title}</h3>
+                        <p className="OneUI-body-text">{description}</p>
+                    </div>
+                    <div {...elem('actions')}>
+                        {actions &&
+                            actions.map((action, index) => (
+                                <React.Fragment key={`${action.text}`}>
+                                    <ActionItem
+                                        action={action}
+                                        dismissToast={() => sonnerToast.dismiss(t)}
+                                    />
+                                    {index < actions.length - 1 && <>•</>}
+                                </React.Fragment>
+                            ))}
+                    </div>
+                </div>
+                {isClosable && (
+                    <button
+                        {...elem('closeButton')}
+                        type="button"
+                        onClick={() => sonnerToast.dismiss(t)}
+                        aria-label={closeButtonLabel}
+                    >
+                        <CloseIcon {...elem('closeIcon')} />
+                    </button>
+                )}
+            </div>
+        ),
+        actions ? { important: true } : {} // focused if actionable
+    );
+
+const bemToaster = bem('Toaster', styles);
+
+export const Toaster = ({ duration = 2500, ...props }) => (
+    <SonnerToaster
+        {...{
+            ...props,
+            duration,
+            closeButton: true,
+            offset: 16,
+            gap: 8,
+            ...bemToaster.block(props),
+        }}
+    />
+);
+
+toast.displayName = 'toast';
```

Comment:
Is it react component? I wondering why it starts from lowercase?

Reply:
It's actually a function! But it is a function we need to call from anywhere, hence the displayName. We use the same structure as the Sonner library: Toaster component, and a toast function to make toast appear

Reply:
hmm, interesting :) so it's kind of imperative way showing notifications.